src/ArticleList.jsx

```diff
@@ -1,15 +1,17 @@
 import { useState, useEffect } from 'react'
+import { useParams } from 'react-router-dom'
 import { fetchArticles } from '../api'
 import ArticleCard from './ArticleCard'
 import Loading from './Loading'
 
 const ArticleList = ({user}) => {
     const [articles, setArticles] = useState([])
     const [isLoading, setIsLoading] = useState(true)
+    const { topic } = useParams()
     
     useEffect(() => {
         setIsLoading(true)
-        fetchArticles().then((articles) => {
+        fetchArticles(topic).then((articles) => {
             if(user){
                 const userArticles = articles.filter((article) => article.author === user)
                 setArticles(userArticles)
```

Comment:
ah yep, it's an optional parameter currently being used to display an article list with only user posts on their profile. This might work better with some conditional rendering in separate component though